Client/src/main/java/de/baleipzig/pdfextraction/client/controller/ImportController.java

```diff
@@ -104,4 +107,78 @@ private void onRequestCompleted(final List<String> name) {
                 .flatMap(templateName -> getLabelMatching(templateName, labels))
                 .ifPresent(l -> templateComboBox.getSelectionModel().select(l));
     }
+
+    public void showTemplateButtonOnAction() {
+
+        if (this.job.getPathToFile() == null) {
+            AlertUtils.showErrorAlert(getResource("alertChoosePDF"));
+            return;
+        }
+
+        if (!showTemplateToggle) {
+
+            String selectedTemplate = this.job.getTemplateName();
+
+            if (selectedTemplate == null) {
+                AlertUtils.showErrorAlert(getResource("alertChooseTemplate"));
+                return;
+            }
+
+            loadTemplate(selectedTemplate);
+            this.showTemplateToggle = true;
+            this.showTemplateButton.setText(getResource("undoShowTemplateButton"));
+        } else {
+
+            removeRectangles();
+            this.showTemplateToggle = false;
+            this.showTemplateButton.setText(getResource("showTemplateButton"));
+        }
+    }
+
+    private void removeRectangles() {
+
+        List<Rectangle> addedRectangles = pdfAnchor.getChildren()
+                .stream()
+                .filter(Rectangle.class::isInstance)
+                .map(node -> (Rectangle) node)
+                .toList();
+
+        pdfAnchor.getChildren().removeAll(addedRectangles);
+    }
+
+    private void loadTemplate(String templateName) {
+
+        this.connector.getForName(templateName)
+                .doOnError(err -> LoggerFactory.getLogger(ImportController.class)
+                        .error("Exception while listening for response.", err))
+                .doOnError(err -> Platform.runLater(() -> AlertUtils.showErrorAlert(err)))
+                .subscribe(templateDTO -> Platform.runLater(() -> drawRectangles(templateDTO)))
+        ;
+    }
+
+    private void drawRectangles(TemplateDTO templateDTO) {
+
+        DrawRectangleWU drawRectangleWU = new DrawRectangleWU(pdfPreviewController.pdfPreviewImageView,
+                templateDTO, renderer.getCurrentPage());
+
+        List<Rectangle> rectangles = drawRectangleWU.work();
+        pdfAnchor.getChildren().addAll(rectangles);
+    }
+
+    private void checkShowTemplateButtonCondition() {
+
+        if (this.job.getTemplateName() != null && this.job.getPathToFile() != null){
+            showTemplateButton.setDisable(false);
+            showTemplateButton.getTooltip().setText(getResource("showTemplateButtonTooltipEnabled"));
+        } else {
+            showTemplateButton.setDisable(true);
+            showTemplateButton.getTooltip().setText(getResource("showTemplateButtonTooltipDisabled"));
+        }
+    }
+
+
+    @Override
+    public void propertyChange(PropertyChangeEvent evt) {
+        checkShowTemplateButtonCondition();
+    }
```

Comment:
Funktionale Interfaces implementiert man nicht, man benutzt sie in lambda's

Reply:
siehe Discord xD